subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/JvmClassHasher.java

```diff
@@ -64,7 +67,8 @@ public HashCode hashClassFile(FileDetails fileDetails) {
             }
             signature = hasher.hash();
         } catch (Exception e) {
-            throw new UncheckedIOException("Could not calculate the signature for class file " + file, e);
+            signature = MALFORMED_CLASS;
```

Comment:
Shouldn't we hash the full content of the class instead? Right now we are ignoring the content completely. That might be wrong if the class is actually valid, but our ASM analysis was broken.